libs/payments/stripe/src/lib/stripe.manager.ts

```diff
@@ -74,4 +76,36 @@ export class StripeManager {
       customer.tax?.automatic_tax === 'not_collecting'
     );
   }
+
+  /**
+   * If customer taxId does not match incoming taxId, update customer record with incoming taxId.
+   *
+   * @param customerId
+   * @param taxId
+   * @returns
+   */
+  async addTaxIdToCustomer(customerId: string, taxId: string) {
+    // fetch customer by customerId
+    const customer = await this.client.fetchCustomer(customerId);
+
+    // check that customer exists
+    if (!customer) {
+      throw new Error('customer does not exist');
+    }
+
+    // check that customer hasn't been deleted
+    if (customer.deleted) throw new CustomerDeletedError();
+
+    // check that customer does not already have taxId
+    if (!customer?.tax_ids && !customer.tax_ids?.object.includes(taxId)) {
```

Comment:
This check should really be put into a getTaxIdForCustomer method that returns the tax ID. That method could then be used here.